### DISCUSSION TITLE
jk bms charging status and control

### DISCUSSION THREAD

**Discussion by riker65** [syssi/esphome-jk-bms#317] [General]
Hi
I was wondering why charging status was on even in control I switched charging of.

I do  not understand, am I doing something wrong?
Or is this an error in the bms firmware?

see : in control charging is off , but in status still on.
![Screenshot_20230426-185307](https://user-images.githubusercontent.com/20332429/234667835-515f7ba9-c29b-430b-b1ef-11360a1f613b.jpg)
![Screenshot_20230426-185314](https://user-images.githubusercontent.com/20332429/234667843-52f99963-ca09-4942-9438-e0b4d13b6693.jpg)

any one experienced the same?

**Comment by syssi** [comment]
(First make sure there is **no** heavy load!!) Could you turn off the `discharging switch` too and measure the voltage between B+ and P-?

**Reply by riker65** [reply]
will check
charing was switched of before the load, because I wanted to drain the cells to 80% capacity and avoid charging by solar

**Comment by riker65** [comment]
I switched discharging off, but still output power from battery 
charging state and control still inconsistant.
I refreshed settings via button 3 to 4 times


![image](https://user-images.githubusercontent.com/20332429/234673880-f17d5bfa-f37b-47a7-8c24-59e875ece1d3.png)



<html>
<body>


20:52:03 | [I] | [jk_bms_ble:997] | Balancer switch: on
-- | -- | -- | --
20:52:03 | [I] | [jk_bms_ble:997] | Balancer switch: on
20:52:03 | [I] | [jk_bms_ble:1001] | Nominal battery capacity: 280.000000 Ah
20:52:03 | [I] | [jk_bms_ble:1007] | Start balance voltage: 3.300000 V


</body>
</html>

**Reply by syssi** [reply]
I will try the same experiment tomorrow. If you have the chance please repeat the experiment while using the Android app only. We shouldnt mix between implementations to avoid confusion.

**Reply by riker65** [reply]
will try also tomorrow

**Comment by riker65** [comment]
I think, charge is changing state to on, weven controll switch charge is off, when load is on BMS.
When load is off, charge state is off again.

Not sure if there is a threshold on the load?

**Reply by syssi** [reply]
> Not sure if there is a threshold on the load?

I don't know any details here.

I made the same test:

1. Android App: If I turn off the charging switch the `Charge` indicator at the main view turns off (no load attached!)
2. Android App: If I turn off the discharging switch the `Discharge` indicator at the main view turns off (no load attached!)
3. ESPHome implementation: If I turn off the `switch.charging` the `binary_sensor.charging` turns off
4. ESPHome implementation: If I turn off the `switch.discharging` the `binary_sensor.discharging` turns off
5. ESPHome implementation: If I turn on the `switch.charging` and cool down the temperature sensor to hit the under-temperature limit the `binary_sensor.charging` should turn off. The switch entity won't change.
6. ESPHome implementation: If I turn on the `switch.discharging` and trigger the undervoltage protection the `binary_sensor.discharging` should turn off. The switch entity won't change.

**Comment by riker65** [comment]
Looks like the strange behaviour I saw was caused by my settings.
I did set OVP very low to avoid overloading and wanted to keep the cells on 80% SOC.
while doing some tests ( playing around) to reduce SOC from 98 to 80 % to rest the cells better, on heavy looad (80 amps) charging turns on, even control switch is off.

So I now unclear if it is a feature or a bug.

still doing some tests so .

**Reply by syssi** [reply]
Off-topic: Please try to avoid switching the mosfets under high load. The chance is high to destroy the mosfets.

**Reply by riker65** [reply]
Off topic: I switched BLE off in esphome implementation, but it switches on again. 
Not sure why and how?   Probably in a reboot.
can the switch state be retained ?
